test/OpenSpaceToolkit/Astrodynamics/Flight/Profile/Tabulated.test.cpp

```diff
@@ -0,0 +1,198 @@
+            states_.add({instant, position, velocity, attitude, angularVelocity, Frame::GCRF()});
+        }
+
+        {
+            const Instant instant = Instant::DateTime(DateTime(2024, 1, 29, 0, 0, 30), Scale::UTC);
+            const Position position =
+                Position::Meters({738650.069144404028, -3587215.226667015813, 5841866.765430007130}, Frame::GCRF());
+            const Velocity velocity =
+                Velocity::MetersPerSecond({-590.931279522597, -6492.770161294917, -3900.690931064452}, Frame::GCRF());
+            const Quaternion attitude = {
+                -0.635371915567, -0.174049521049, 0.723374654124, 0.206732684649, Quaternion::Format::XYZS
+            };
+            const Vector3d angularVelocity = {0.0, 0.0, 0.0};
+            states_.add({instant, position, velocity, attitude, angularVelocity, Frame::GCRF()});
+        }
+    }
+
+    Array<State> states_ = Array<State>::Empty();
+};
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_Profile_Models_Tabulated, Constructor)
+{
+    {
+        EXPECT_NO_THROW(Tabulated tabulated(states_););
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_Profile_Models_Tabulated, EqualToOperator)
+{
+    const Tabulated tabulated = {states_};
+
+    {
+        EXPECT_TRUE(tabulated == tabulated);
+    }
+
+    {
+        const Array<State> statesSubset = {states_[0]};
+        const Tabulated tabulatedSubset = {statesSubset};
+        EXPECT_FALSE(tabulated == tabulatedSubset);
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_Profile_Models_Tabulated, NotEqualToOperator)
+{
+    const Tabulated tabulated = {states_};
+
+    {
+        EXPECT_FALSE(tabulated != tabulated);
+    }
+
+    {
+        const Array<State> statesSubset = {states_[0]};
+        const Tabulated tabulatedSubset = {statesSubset};
+        EXPECT_TRUE(tabulated != tabulatedSubset);
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_Profile_Models_Tabulated, StreamOperator)
+{
+    {
+        const Tabulated tabulated = {states_};
+
+        testing::internal::CaptureStdout();
+
+        EXPECT_NO_THROW(std::cout << tabulated << std::endl);
+
+        EXPECT_FALSE(testing::internal::GetCapturedStdout().empty());
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_Profile_Models_Tabulated, IsDefined)
+{
+    {
+        const Tabulated tabulated = {states_};
+        EXPECT_TRUE(tabulated.isDefined());
+    }
+
+    {
+        const Tabulated tabulated = {{}};
+        EXPECT_FALSE(tabulated.isDefined());
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_Profile_Models_Tabulated, Getters)
+{
+    {
+        const Tabulated tabulated = {states_};
+        const Interval interval = tabulated.getInterval();
+        EXPECT_EQ(interval.getStart(), states_.accessFirst().accessInstant());
+        EXPECT_EQ(interval.getEnd(), states_.accessLast().accessInstant());
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_Profile_Models_Tabulated, CalculateStateAt)
```

Comment:
Would be good to add more validation style tests (reading in a file of quaternions/angvels generated from GMAT etc) for this function in the future

Reply:
Yeah outside the scope of this MR.